livre/Eq/Window.cpp

```diff
@@ -91,22 +91,23 @@ class Window::Impl
         node->getDashTree()->getRenderStatus().setThreadOp( TO_EXIT );
     }
 
-    void frameStart( const uint32_t frameNumber )
+    void frameStart()
     {
         _dashProcessorPtr->getDashContext()->setCurrent();
+    }
 
+    void frameFinish( const uint32_t frameNumber )
+    {
         livre::Node* node = static_cast< livre::Node* >( _window->getNode( ));
         DashRenderStatus& renderStatus = node->getDashTree()->getRenderStatus();
 
         Pipe* pipe = static_cast< Pipe* >( _window->getPipe( ));
-        const uint32_t startFrame = pipe->getFrameData()->getAppParameters()->frames.x();
-
         if( pipe->getFrameData()->getAppParameters()->animationEnabled )
-            renderStatus.setFrameID( startFrame + frameNumber - 1 );
-        else
-            renderStatus.setFrameID( startFrame );
-
-        commit();
+        {
+            const Vector2ui& animationFrames = pipe->getFrameData()->getAppParameters()->frames;
+            const uint32_t duration = animationFrames.y() - animationFrames.x();
+            renderStatus.setFrameID( animationFrames.x() + ( frameNumber % duration ));
```

Comment:
It is if the number of maxFrames is larger than the frame range for the animation.